public/js/controllers/blocks.js

```diff
@@ -0,0 +1,314 @@
+        fees += tx.Fees;
+    }
+
+    return `<div class="block">
+                <div class="block-info">
+                    <a class="color-code" href="/mempool">Mempool</a>
+                    <div class="mono" style="line-height: 1;">${Math.floor(block.Total)} DCR</div>
+                    <span class="timespan">
+                        <span data-target="main.age" data-age="${block.Time}"></span>&nbsp;ago
+                    </span>
+                </div>
+                <div class="block-rows">
+                    ${makeRewardsElement(mempoolSubsidy, fees, block.Votes.length, '#')}
+                    ${makeVoteElements(block.Votes)}
+                    ${makeTicketAndRevocationElements(block.Tickets, block.Revocations, '/mempool')}
+                    ${makeTransactionElements(block.Transactions, '/mempool')}
+                </div>
+            </div>`;
+}
+
+function newBlockHtmlElement(block) {
+    let rewardTxId;
+    for (const tx of block.Transactions) {
+        if (tx.Coinbase) {
+            rewardTxId = tx.TxID;
+            break;
+        }
+    }
+
+    return `<div class="block">
+                ${makeBlockSummary(block.Height, block.TotalSent, block.Time)}
+                <div class="block-rows">
+                    ${makeRewardsElement(block.Subsidy, block.MiningFee, block.Votes.length, rewardTxId)}
+                    ${makeVoteElements(block.Votes)}
+                    ${makeTicketAndRevocationElements(block.Tickets, block.Revocations, `/block/${block.Height}`)}
+                    ${makeTransactionElements(block.Transactions, `/block/${block.Height}`)}
+                </div>
+            </div>`;
+}
+
+function makeBlockSummary(blockHeight, totalSent, time) {
+    return `<div class="block-info">
+                <a class="color-code" href="/block/${blockHeight}">${blockHeight}</a>
+                <div class="mono" style="line-height: 1;">${Math.floor(totalSent)} DCR</div>
+                <span class="timespan">
+                    <span data-target="main.age" data-age="${time}"></span>&nbsp;ago
+                </span>
+            </div>`;
+}
+
+function makeRewardsElement(subsidy, fee, voteCount, rewardTxId) {
+    if (!subsidy) {
+        return `<div class="block-rewards">
+                    <span class="pow"><span class="paint" style="width:100%;"></span></span>
+                    <span class="pos"><span class="paint" style="width:100%;"></span></span>
+                    <span class="fund"><span class="paint" style="width:100%;"></span></span>
+                    <span class="fees" title='{"object": "Tx Fees", "total": "${fee}"}'></span>
+                </div>`;
+    }
+
+    const pow = subsidy.pow / conversionRate;
+    const pos = subsidy.pos / conversionRate;
+    const fund = (subsidy.developer || subsidy.dev) / conversionRate;
+    
+    const backgroundColorRelativeToVotes = `style="width: ${voteCount * 20}%"`; // 5 blocks = 100% painting
```

Comment:
This shrinks all the reward divs when votes are missing, that seems off to me.
What's the intent here?
<img width="625" alt="bg" src="https://user-images.githubusercontent.com/25571523/47415772-1aea6500-d729-11e8-8617-7cc67a9592e3.png">

Reply:
The updated PR description explains the intent. The partial background filling is a factor of the number of votes in the block. The background currently fills 4/5th the available width for each cell because there are 4 votes in the block

Reply:
Oh interesting, I didn't realize the dev fund also gets reduced when there's less than 5 tickets! Proof is in the pudding, visual blocks are educational!